### DISCUSSION TITLE
Documentation of next-i18next for Typescript

### DISCUSSION THREAD

**Discussion by pramod-bhandarkar** [i18next/next-i18next#1447] [General]
Hi All,

Apologies if this is duplicate of old issues or any discussion. 

I am new to react and very new to Next JS app development. I want build the app with translation support. Exploring on available libraries and after checking on react-i18next i was landed in project of next-i18next. 

I am unable to find any documentation for configure this with typescript project. I have explored many issues raised on this topic but there was no straight forward answer i could find. I know i am very late to the party so please help where i can find an example or a documentation with which i can setup next-i18next for a typescript codebase.

I tried this link but it is just files with tsx extension nothing different compared to Next JS javascript build. https://github.com/isaachinman/next-i18next-typescript

@isaachinman  Could you please guide me to a proper discussion or issue which helps me with a documentation. 

Thanks.

**Comment by isaachinman** [comment]
Sorry, what exactly do you need in terms of documentation? `next-i18next` is written in TS entirely and exposes types, but you don't need to do anything special between JS or TS.

**Reply by pramod-bhandarkar** [reply]
Hi @isaachinman Thanks, I could figure out by looking into exported types on how to work with library. Page errors are disappeared now. However, I am getting this strange error when I run build of the next project.

It will be great if I get some help here or some directions. 

```
info  - Collecting page data  
[==  ] info  - Generating static pages (8/15)Unhandled error during request: Error: Cannot find module 'D:\my\apps\my-app-dash\next-i18next.config.js'
    at webpackEmptyContext (D:\my\apps\my-app-dash\.next\serverless\pages\btwifi.js:8:10)
    at D:\my\apps\my-app-dash\.next\serverless\chunks\324.js:1231:72 {
  code: 'MODULE_NOT_FOUND'
}
Error occurred prerendering page "/en/dash". Read more: https://nextjs.org/docs/messages/prerender-error
```

My file is right there in the path "D:\my\apps\my-app-dash\next-i18next.config.js"

Build will be successful if I remove the getstatprops method.

`export const getStaticProps = async ({
  locale,
}: PathProps): Promise<{ props: SSRConfig }> => {
  return {
    props: {
      ...(await serverSideTranslations(locale, ['common', 'home', 'faq'])),
    },
  }
}`

 also there are quite a few warnings.

./node_modules/next/dist/server/load-components.js
Critical dependency: the request of a dependency is an expression

./node_modules/next/dist/server/load-components.js
Critical dependency: the request of a dependency is an expression

./node_modules/next/dist/server/load-components.js
Critical dependency: the request of a dependency is an expression

./node_modules/next/dist/server/require.js
Critical dependency: the request of a dependency is an expression

./node_modules/next/dist/server/require.js
Critical dependency: the request of a dependency is an expression

./node_modules/next/dist/server/require.js
Critical dependency: the request of a dependency is an expression

./node_modules/i18next-fs-backend/cjs/readFile.js
Critical dependency: the request of a dependency is an expression

./node_modules/i18next-fs-backend/cjs/readFile.js
Critical dependency: the request of a dependency is an expression

**1. Is it mandatory to use getStaticProps or getServerSideProps if i want client side rendering only?**

**Reply by isaachinman** [reply]
Please see #1202.

**Reply by pramod-bhandarkar** [reply]
Thanks @isaachinman for the prompt response. I will check issue #1202.

**Reply by pramod-bhandarkar** [reply]
the solution to pass the config as an argument worked for me. However, I will keep the webpack at version 5 only as I do not want to downgrade my nextjs version from 11.1.0. Thanks @isaachinman

**Reply by pramod-bhandarkar** [reply]
hi @isaachinman sorry to bother you again but I forgot to mention this thing. My app runs on a serverless architecture and my local build worked just fine but when i deploy it i am seeing 503 error on aws serverless environment. is there any workaround to make this work on serverless architecture?

**Reply by isaachinman** [reply]
You can search through previous issues to find plenty of discussion about deploying on serverless platforms. Yes – it's possible.
